shared/misc/upload_ansible_roles_to_galaxy.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python2
+
+from tempfile import mkdtemp
+import os
+import os.path
+import sys
+import shutil
+import re
+import argparse
+import getpass
+
+try:
+    from github import Github
+except ImportError:
+    sys.stderr.write("Please install PyGithub, on Fedora it's in the "
+                     "python-PyGithub package.\n")
+    sys.exit(1)
+
+
+def create_empty_repositories(github_new_repos, github_org):
+    for github_new_repo in github_new_repos:
+        print "Creating new Github repository: " + github_new_repo
+        github_org.create_repo(
+            github_new_repo,
+            description="Role generated by OpenSCAP SSG",
+            homepage="https://www.open-scap.org/",
+            private=False,
+            has_issues=False,
+            has_wiki=False,
+            has_downloads=False)
+
+
+def clone_and_init_repositories(repositories_to_create):
+    for repo in repositories_to_create:
+        os.system(
+            "git clone git@github.com:OpenSCAP-Ansible-Roles/%s.git" % repo)
+        os.system("ansible-galaxy init " + repo + " --force")
+        os.chdir(repo)
+        try:
+            os.system('git add .')
+            os.system('git commit -a -m "Initial commit"')
+            os.system('git push origin master')
+        finally:
+            os.chdir("..")
+
+
+def update_repository(repository, local_file_path, meta_template_path):
+    print("Processing %s..." % repository.name)
+
+    with open(local_file_path, 'r') as f:
+        filedata = f.read()
+
+    local_file_content = filedata.replace("   tasks:", "#   tasks:")
+    remote_file = repository.get_file_contents("/tasks/main.yml")
+
+    if local_file_content != remote_file.decoded_content:
+        repository.update_file(
+            "/tasks/main.yml",
+            "Updates tasks/main.yml",
+            local_file_content,
+            remote_file.sha)
+        print "Updating tasks/main.yml in " + repository.name
+
+    separator = "#" * 79
+    header = filedata[filedata.find(
+        separator) + len(separator) + 3:filedata.rfind(separator) - 3]
+    header = header.replace('# ', '')
+    header = header.replace('#', '')
+    title = re.search('Profile Title:\s+(.+)$', header, re.MULTILINE).group(1)
+    header = header.replace('\n', '  \n')
```

Comment:
I would improve this section, I don't like the the indices of `filedata` list are counted (`+3`, `-3`, ...). I also don't like how the spaces and newlines are replaced.

Reply:
yeah, this part is pretty bad, I kind of know what's going on here but it's very hackish...

Reply:
I also realized just now that it's using some very questionable assumptions. We should look for the first and second occurence of the separator, not for the first and last. I will rewrite this I think, it can be done via one regex pretty reliable I think.